gelf/reader.go

```diff
@@ -139,6 +157,41 @@ func (r *Reader) ReadMessage() (*Message, error) {
 	return msg, nil
 }
 
+func getDeframentator(r *Reader, messageID string) *defragmentator {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	defragmentator, ok := r.messageDefragmentators[messageID]
+	if !ok {
+		defragmentator = newDefragmentator(r.maxMessageTimeout)
+		r.messageDefragmentators[messageID] = defragmentator
+	}
+	return defragmentator
+}
+
 func (r *Reader) Close() error {
+	close(r.done)
 	return r.conn.Close()
 }
+
+func (r *Reader) initDefragmentatorsCleanup(defarmentatorsCleanUpInterval time.Duration) {
+	go func() {
+		for {
+			select {
+			case <-r.done:
+				return
+			case <-time.After(defarmentatorsCleanUpInterval):
+				cleanUpExpiredDefragmentators(r)
+			}
+		}
+	}()
+}
+
+func cleanUpExpiredDefragmentators(r *Reader) {
```

Comment:
Nit: Not too sure this feels more consistent

```suggestion
func (r *Reader)cleanUpExpiredDefragmentators() {
```

Reply:
done